test/extensions/clusters/common/logical_host_integration_test.cc

```diff
@@ -20,27 +20,29 @@ class LogicalHostIntegrationTest : public testing::Test, public HttpIntegrationT
 
   NiceMock<Network::MockDnsResolverFactory> dns_resolver_factory_;
   Registry::InjectFactory<Network::DnsResolverFactory> registered_dns_factory_;
+  std::shared_ptr<uint32_t> address_ptr_ = nullptr;
 };
 
 // Reproduces a race from https://github.com/envoyproxy/envoy/issues/32850.
 // The test is by mocking the DNS resolver to return multiple different
 // addresses, also config dns_refresh_rate to be extremely fast.
 TEST_F(LogicalHostIntegrationTest, LogicalDNSRaceCrashTest) {
+  address_ptr_ = std::make_shared<uint32_t>(0);
   auto dns_resolver = std::make_shared<Network::MockDnsResolver>();
   EXPECT_CALL(dns_resolver_factory_, createDnsResolver(_, _, _))
       .WillRepeatedly(testing::Return(dns_resolver));
   EXPECT_CALL(*dns_resolver, resolve(_, _, _))
       .WillRepeatedly(
-          Invoke([&](const std::string&, Network::DnsLookupFamily,
+          Invoke([address_ptr = address_ptr_](
```

Comment:
and thne you can just pass address_ptr here

Reply:
done